.github/workflows/validate-demo-metadata.yml

```
@@ -0,0 +1,34 @@
+# Starting as a separate Action as a POC.
+# Consider combining this with the build-pr
+# Action if we want to keep it.
+name: Validate Demo Metadata
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - dev
+
```

Let's add a concurrency group here as well so older buids are auto-cancelled.

```suggestion
concurrency:
  group: metadata-validation-${{ github.ref }}
  cancel-in-progress: true
```